### PR TITLE
pm: add pm service support for ish

### PR DESCRIPTION
**Pull request by leifuzhao**
This enables the power management for Intel ISH. It supports D0i0, D0i1,
D0i2, D0i3 power saving modes for ISH.

### CODE REVIEW

**Comment by kwd-doodling** [comment]
add domain to commit titles.
add "x86:" to "linker: add linker script for ish aon section".
add "soc: ish:" to "pm: add pm service support for ish)"

**Review by ceolin — DISMISSED** [review]
Looks good to me, minor comment.

File: west.yml [hunk 1]
```diff
@@ -168,7 +168,7 @@ manifest:
       groups:
         - hal
     - name: hal_intel
-      revision: 89a0db5d9155484624447d62a5211bcf6f6f0387
+      revision: af547115d1c1e125f096a440f9583866a01c1bcf
```
Comment (nashif) [inline west.yml:171]: what is this sha? It Can't find it anywhere.

**Review by nashif — CHANGES_REQUESTED** [review]

File: west.yml (hunk 1 above)
Comment (leifuzhao) [reply west.yml:171]: it is from https://github.com/intel/hal-intel/pull/20.

Comment (kwd-doodling) [reply west.yml:171]: will update it after https://github.com/intel/hal-intel/pull/20 merged.

**Review by kwd-doodling — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by ceolin — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**